mdata/cache/ccache_test.go

```diff
@@ -502,3 +502,99 @@ func testSearchDisconnectedWithGapStartEnd(t *testing.T, spanaware, ascending bo
+	for _, itgen := range itgens {
+		cc.Add(metric1_1, rawMetric1, 0, itgen)
+		cc.Add(metric1_2, rawMetric1, 0, itgen)
+		cc.Add(metric2_1, rawMetric2, 0, itgen)
+	}
+
+	// check if Search returns them all for metric1_1
+	res = cc.Search(test.NewContext(), metric1_1, 1000, uint32(1000+itgenCount*len(values)))
+	if len(res.Start) != itgenCount {
+		t.Fatalf("Expected to have %d values, got %d", itgenCount, len(res.Start))
+	}
+
+	// check if Search returns them all for metric1_2
+	res = cc.Search(test.NewContext(), metric1_2, 1000, uint32(1000+itgenCount*len(values)))
+	if len(res.Start) != itgenCount {
+		t.Fatalf("Expected to have %d values, got %d", itgenCount, len(res.Start))
+	}
+
+	// check if Search returns them all for metric2_1
+	res = cc.Search(test.NewContext(), metric2_1, 1000, uint32(1000+itgenCount*len(values)))
+	if len(res.Start) != itgenCount {
+		t.Fatalf("Expected to have %d values, got %d", itgenCount, len(res.Start))
+	}
+
+	// now delete metric1_1, but leave metric2_1
+	delRes := cc.DelMetric(rawMetric1)
+	expectDelSeries := 1
+	if delRes.Series != expectDelSeries {
+		t.Fatalf("Expected exactly %d series to get deleted, but got %d", expectDelSeries, delRes.Series)
+	}
+
+	expectDelArchives := 2
+	if delRes.Archives != expectDelArchives {
+		t.Fatalf("Expected exactly %d archives to get deleted, but got %d", expectDelArchives, delRes.Archives)
+	}
+
+	// check if metric1_1 returns no results anymore
+	res = cc.Search(test.NewContext(), metric1_1, 1000, uint32(1000+itgenCount*len(values)))
+	if len(res.Start) != 0 {
+		t.Fatalf("Expected to have %d values, got %d", 0, len(res.Start))
+	}
+
+	// check if metric1_1 returns no results anymore
+	res = cc.Search(test.NewContext(), metric1_2, 1000, uint32(1000+itgenCount*len(values)))
+	if len(res.Start) != 0 {
+		t.Fatalf("Expected to have %d values, got %d", 0, len(res.Start))
+	}
+
+	// but metric2_1 should still be there
+	res = cc.Search(test.NewContext(), metric2_1, 1000, uint32(1000+itgenCount*len(values)))
+	if len(res.Start) != itgenCount {
+		t.Fatalf("Expected to have %d values, got %d", itgenCount, len(res.Start))
+	}
+
+	// now add metric1_1 and metric1_2 again
+	for _, itgen := range itgens {
+		cc.Add(metric1_1, rawMetric1, 0, itgen)
+		cc.Add(metric1_2, rawMetric1, 0, itgen)
+	}
+
+	// and check if it gets returned by Search again
+	res = cc.Search(test.NewContext(), metric1_1, 1000, uint32(1000+itgenCount*len(values)))
+	if len(res.Start) != itgenCount {
+		t.Fatalf("Expected to have %d values, got %d", itgenCount, len(res.Start))
+	}
+
+	// now reset the whole metric cache
+	delRes = cc.Reset()
+	expectDelSeries = 2
+	if delRes.Series != expectDelSeries {
+		t.Fatalf("Expected exactly %d series to get deleted, but got %d", expectDelSeries, delRes.Series)
+	}
+
+	expectDelArchives = 3
+	if delRes.Archives != expectDelArchives {
+		t.Fatalf("Expected exactly %d archives to get deleted, but got %d", expectDelArchives, delRes.Archives)
+	}
```

Comment:
i would like to see after a reset whether the cache behaves properly (e.g. whether adds and searches and deletes still work). that would be out of scope for TestMetricDelete though (and perhaps it can be implemented by a TestReset test function which calls the other test functions for adds, deletes, etc so it doesn't duplicate that logic)

Reply:
how's that: https://github.com/grafana/metrictank/pull/555/commits/a406e734049abd2daa4cc3422f969b6b936fd58d

Reply:
looks good :+1: